common/include/pcl/pcl_macros.h

```diff
@@ -128,9 +128,14 @@
  *
  * \param x A parameter which will be used without being evaluated.
  *
- * Slightly better than `(void)x` as explained [here](https://stackoverflow.com/a/4030983/9926122).
+ * Slightly better than `(void)(x)` as explained [here](https://stackoverflow.com/a/4030983/9926122)
+ * but only works on MSVC so fall back on the classic version for other cases.
  */
-#define PCL_UNUSED(x) ((void)(true ? 0 : ((x), void(), 0)))
+#if defined _MSC_VER
```

Comment:
What about `template<class T> void ignore( const T& ) { }` instead? This should work for cases just like cast to `void` and isn't a macro? It works for all compilers too (for obv reasons, since this is like kicking the can a bit further)

Reply:
It's going to be compiled for every `T`, though. I don't know, I never use macros myself if I can (and it shows), but that'd be a lot of instantiations, right?

Reply:
Not a lot of uses anyways, right? I searched for `(void)` and got 77 total uses (not enough to count for a library like PCL).

BTW, does the `(void)x` version work for MSVC? If it does, we can just use the original `PCL_UNUSED` without complicating this

Reply:
I don't like macros and I stand by it, so I'm moving to the template function, but `git grep -n "(void)"` gives 319 results to me (albeit 167 being from opennurbs and poisson4 things).

Reply:
saw what you meant